### PR TITLE
Changes Various defines to an Enum

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Description
#5839 but for master.

## **Discord contact info**
AsparagusEduardo

### CODE REVIEW

File: docs/tutorials/how_to_battle_script_command_macro.md [hunk 1]
```diff
@@ -17,7 +17,17 @@ In general, `gBattlescriptCurrInstr` tracks the current battle script position a
...
+For example:
+```diff
+enum CmdVarious
+{
+    ...
+    VARIOUS_SUCKER_PUNCH_CHECK,
++   VARIOUS_SET_SIMPLE_BEAM,
+    VARIOUS_TRY_ENTRAINMENT,
+    ...
+};
```
Comment (AlexOn1ine) [inline docs/tutorials/how_to_battle_script_command_macro.md:29]: Should we promote adding variouses instead of callnatives?

Comment (Bassoonian) [reply docs/tutorials/how_to_battle_script_command_macro.md:29]: Couldn't agree more with Alex here

Comment (AsparagusEduardo) [reply docs/tutorials/how_to_battle_script_command_macro.md:29]: Also agree, did this out of habit of updating docs.
I'll change the tutorial to account for this.

File: docs/tutorials/how_to_battle_script_command_macro.md [hunk 2]
```diff
@@ -1,51 +1,47 @@
 ## How to add new Battle Script Commands/Macros
 
-To preface this tutorial, the battle engine upgrade has exhausted all battle script command IDs, and instead uses the `various` command to effectively add new commands. This is preferential to creating a secondary battle script command table like is done in the CFRU.
+To preface this tutorial, the battle engine upgrade has exhausted all battle script command IDs. Historically, we've used the `various` command to effectively add new commands. However, this has caused issues of mantainability and readability due to the massive switch needed for it. Thanks to the cleanup made by the team and contributors, we now are able to call an infinite amount of commands by using `callnative`, This is preferential to creating a secondary battle script command table like is done in the CFRU.
```
Comment (Bassoonian) [inline docs/tutorials/how_to_battle_script_command_macro.md:3]:
```suggestion
To preface this tutorial, the battle engine upgrade has exhausted all battle script command IDs. Historically, we've used the `various` command to effectively add new commands. However, this has caused issues of maintainability and readability due to the massive switch needed for it. Thanks to the cleanup made by the team and contributors, we now are able to call an infinite amount of commands by using `callnative`. This is preferential to creating a secondary battle script command table like is done in the CFRU.
```

**Review by Bassoonian — CHANGES_REQUESTED** [review]

File: docs/tutorials/how_to_battle_script_command_macro.md [hunk 3]
```diff
@@ -1,51 +1,47 @@
...
-    .byte 0x76
-    .byte \battler
-    .byte \param1
-    .endm
+	.macro callnative func:req
+	.byte 0xff
+	.4byte \func
+	.endm
 ```
+`callnative` uses the last battle script command ID in order to pass a native function as an argument. Arguments are added recursively via a macro, so no need to worry about how they need to align to the amount of instructions to skip.
```
Comment (AlexOn1ine) [inline docs/tutorials/how_to_battle_script_command_macro.md:13]: maybe add that arguments are optional?

Comment (equezada-jej) [reply docs/tutorials/how_to_battle_script_command_macro.md:13]:
```suggestion
`callnative` uses the last battle script command ID in order to pass a native function as an argument. Additional optional arguments are added recursively via a macro, so no need to worry about how they need to align to the amount of instructions to skip. H
```

Comment (AsparagusEduardo) [reply docs/tutorials/how_to_battle_script_command_macro.md:13]:
```suggestion
`callnative` uses the last battle script command ID in order to pass a native function as an argument. Additional optional arguments are added recursively via a macro, so no need to worry about how they need to align to the amount of instructions to skip.
```

**Review by AlexOn1ine — APPROVED** [review]

**Comment by AlexOn1ine** [comment]
@Bassoonian can be merged from my side

**Review by Bassoonian — APPROVED** [review]

**Pull request merged**